### PR TITLE
Update Tentacle to allow it to register itself as a KubernetesTentacle

### PR DESCRIPTION
**Pull request by scme0**
# Background

Part of #project-k8s-agent

# Results

Updating Tentacle to be able to register itself as a Kubernetes Tentacle. This adds a new command rather than updating the `register-with` command but most of the functionality from `register-with` is brought across to the new command via inheritance.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: docker/linux/scripts/configure-tentacle.sh [hunk 1]
```diff
@@ -27,203 +27,215 @@ if [ -f "$alreadyConfiguredSemaphore" ]; then
...
-    echo " - tenant tag '$TargetTenantTag'"
-  fi
-  if [[ ! -z "$TargetTenantedDeploymentParticipation" ]]; then
-    echo " - tenanted deployment participation '$TargetTenantedDeploymentParticipation'"
-  fi
-  if [[ ! -z "$Space" ]]; then
-    echo " - space '$Space'"
-  fi
+    if [[ ! -z "$ServerCommsAddress" || ! -z "$ServerPort" ]]; then
+        if [[ ! -z $AsKubernetesTentacle ]]; then
```
Comment (scme0) [inline docker/linux/scripts/configure-tentacle.sh:79]: Currently all you need to do is set the `AsKubernetesTentacle` environment variable to something for  Tentacle to register itself as a tentacle. This script still sets `TentacleActive` or `TentaclePassive` communication styles for the type of connection.

**Comment by scme0** [comment]
[sc-63133]

File: source/Octopus.Manager.Tentacle/TentacleConfiguration/TentacleManager/TentacleManagerModel.cs [hunk 2]
```diff
@@ -166,7 +167,8 @@ void Reload(ApplicationInstanceRecord applicationInstance)
             }
             else
             {
-                var listens = tencon.TrustedOctopusServers.Where(t => t.CommunicationStyle == CommunicationStyle.TentaclePassive).ToList();
+                var listens = tencon.TrustedOctopusServers.Where(t => t.CommunicationStyle == CommunicationStyle.TentaclePassive ||
+                    (t.CommunicationStyle == CommunicationStyle.KubernetesTentacle && t.AgentCommunicationMode == TentacleCommunicationModeResource.Listening)).ToList();
```
Comment (APErebus) [inline source/Octopus.Manager.Tentacle/TentacleConfiguration/TentacleManager/TentacleManagerModel.cs:171]: Looks like a rogue `AgentCommunicationMode` property still floating around?

File: source/Octopus.Tentacle/Configuration/OctopusServerConfiguration.cs [hunk 3]
```diff
@@ -42,6 +43,8 @@ public string Thumbprint
         [JsonConverter(typeof(CommunicationStyleConverter))]
         public CommunicationStyle CommunicationStyle { get; set; }
 
+        public TentacleCommunicationModeResource AgentCommunicationMode { get; set; } = TentacleCommunicationModeResource.Polling;
```
Comment (APErebus) [inline source/Octopus.Tentacle/Configuration/OctopusServerConfiguration.cs:46]:
```suggestion
        public TentacleCommunicationModeResource KubernetesCommunicationMode { get; set; } = TentacleCommunicationModeResource.Polling;
```

**Review by APErebus — CHANGES_REQUESTED** [review]
Looks good! Just the name of one of the properties I think (just to move away from Agent)

File: source/Octopus.Manager.Tentacle/TentacleConfiguration/TentacleManager/TentacleManagerModel.cs (hunk 2 above)
Comment (scme0) [reply source/Octopus.Manager.Tentacle/TentacleConfiguration/TentacleManager/TentacleManagerModel.cs:171]: Ah damn, good catch. Yep I'll fix that up 👍

**Review by APErebus — APPROVED** [review]

**Pull request merged**